app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionDisplayProfiles.scala

```diff
@@ -0,0 +1,168 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.cgtpropertydisposalsstubs.controllers
+import cats.implicits._
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Result
+import play.api.mvc.Results.BadRequest
+import uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.BusinessPartnerRecordController.DesErrorResponse
+import uk.gov.hmrc.cgtpropertydisposalsstubs.controllers.SubscriptionController.{DesSubscriptionDisplayDetails, SubscriptionDetails}
+import uk.gov.hmrc.cgtpropertydisposalsstubs.models.{DesAddressDetails, DesContactDetails, DesIndividual, DesTrustee}
+
+case class SubscriptionDisplay(
+  predicate: String => Boolean,
+  subscriptionDisplayResponse: Either[Result, DesSubscriptionDisplayDetails]
+)
+
+object SubscriptionDisplayProfiles {
+
+  val individualSubscriptionDisplayDetails = DesSubscriptionDisplayDetails(
+    regime = "CGT",
+    subscriptionDetails = SubscriptionDetails(
+      individual = Some(
+        DesIndividual(
+          "Individual",
+          "Luke",
+          "Bishop"
+        )
+      ),
+      None,
+      true,
+      DesAddressDetails(
+        "100 Sutton Street",
+        Some("Wokingham"),
+        Some("Surrey"),
+        Some("London"),
+        "DH14EJ",
+        "GB"
+      ),
+      DesContactDetails(
+        "Stephen Wood",
+        Some("(+013)32752856"),
+        Some("(+44)7782565326"),
+        Some("01332754256"),
+        Some("stephen@abc.co.uk")
+      )
+    )
+  )
+
+  val trusteeSubscriptionDisplayDetails = DesSubscriptionDisplayDetails(
+    regime = "CGT",
+    subscriptionDetails = SubscriptionDetails(
+      None,
+      trustee = Some(
+        DesTrustee(
+          "Trustee",
+          "ABC Trust"
+        )
+      ),
+      true,
+      DesAddressDetails(
+        "101 Kiwi Street",
+        None,
+        None,
+        Some("Christchurch"),
+        "",
+        "NZ"
+      ),
+      DesContactDetails(
+        "Stephen Wood",
+        Some("(+013)32752856"),
+        Some("(+44)7782565326"),
+        Some("01332754256"),
+        Some("stephen@abc.co.uk")
+      )
+    )
+  )
+
+  def getDisplayDetails(id: String): Option[SubscriptionDisplay] =
+    subscriptionDisplayProfiles.find(_.predicate(id))
+
+  private val subscriptionDisplayProfiles = List(
+    SubscriptionDisplay(
+      _ === "XACGTP123456789",
+      Right(individualSubscriptionDisplayDetails)
+    ),
+    SubscriptionDisplay(
+      _ === "XACGTP123456790",
+      Right(trusteeSubscriptionDisplayDetails)
+    ),
+    SubscriptionDisplay(
+      id => id.startsWith("ERG400"),
```

Comment:
the form of a CGT reference number is given by the regex `^(X[A-Z]CGTP[0-9]{9})$` - could we use realistic looking CGT reference numbers in the mapping for error responses?

Reply:
Yes we could. I was following the same design you did for the subscription profiles c.f https://github.com/hmrc/cgt-property-disposals-stubs/blob/e23b210c17acf0751d8b9e5bed094ad4a9f5e8cc/app/uk/gov/hmrc/cgtpropertydisposalsstubs/controllers/SubscriptionProfiles.scala#L120

Reply:
changed